bentoml/deployment/sagemaker/__init__.py

```diff
@@ -343,27 +358,216 @@ def delete(self):
             return False
 
 
-# Deployment Service MVP Working-In-Progress
+class TemporarySageMakerContent(object):
+    def __init__(self, archive_path, bento_name, bento_version, _cleanup=True):
+        self.archive_path = archive_path
+        self.bento_name = bento_name
+        self.bento_version = bento_version
+        self.temp_directory = TempDirectory()
+        self._cleanup = _cleanup
+        self.path = None
+
+    def __enter__(self):
+        self.generate()
+        return self.path
+
+    def generate(self):
+        self.temp_directory.create()
+        tempdir = self.temp_directory.path
+        saved_path = os.path.join(tempdir, self.bento_name, self.bento_version)
+        shutil.copytree(self.archive_path, saved_path)
+
+        with open(os.path.join(saved_path, "nginx.conf"), "w") as f:
+            f.write(DEFAULT_NGINX_CONFIG)
+        with open(os.path.join(saved_path, "wsgi.py"), "w") as f:
+            f.write(DEFAULT_WSGI_PY)
+        with open(os.path.join(saved_path, "serve"), "w") as f:
+            f.write(DEFAULT_SERVE_SCRIPT)
+
+        # permission 755 is required for entry script 'serve'
+        permission = "755"
+        octal_permission = int(permission, 8)
+        os.chmod(os.path.join(saved_path, "serve"), octal_permission)
+        self.path = saved_path
+
+    def cleanup(self, ignore_errors=False):
+        self.temp_directory.cleanup()
+        self.path = None
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self._cleanup:
+            self.cleanup()
+
+
 class SageMakerDeploymentOperator(DeploymentOperatorBase):
     def apply(self, deployment_pb, repo):
-        # deploy code.....
-        spec = deployment_pb.spec
-        bento_path = repo.get(spec.bento_name, spec.bento_version)
+        deployment_spec = deployment_pb.spec
+        sagemaker_config = deployment_spec.sagemaker_operator_config
+        if sagemaker_config is None:
+            raise BentoMLDeploymentException('Sagemaker configuration is missing.')
+
+        region = sagemaker_config.region or config.get('aws', 'default_region')
```

Comment:
think the second case has been handled above. do we still need it here?

Reply:
na, we don't need it here.  It should be handled at CLI entry point